src/data-access/repositories/mongo/mongo-show-style-variant-configuration-listener.ts

```diff
@@ -0,0 +1,52 @@
+import { BaseMongoRepository } from './base-mongo-repository'
+import { DataChangedListener } from '../interfaces/data-changed-listener'
+import { ShowStyleVariant } from '../../../model/entities/show-style-variant'
+import { Logger } from '../../../logger/logger'
+import { MongoDatabase } from './mongo-database'
+import { ChangeStream, ChangeStreamDocument, ChangeStreamOptions } from 'mongodb'
+import { UnsupportedOperationException } from '../../../model/exceptions/unsupported-operation-exception'
+
+const SHOW_STYLE_VARIANT_CONFIGURATION_COLLECTION_NAME: string = 'showStyleVariants'
+
+export class MongoShowStyleVariantConfigurationListener extends BaseMongoRepository implements DataChangedListener<ShowStyleVariant> {
+
+  private readonly logger: Logger
+
+  private onUpdatedCallback: (showStyleVariant: ShowStyleVariant) => void
+
+  constructor(mongoDatabase: MongoDatabase, logger: Logger) {
+    super(mongoDatabase)
+    this.logger = logger.tag(MongoShowStyleVariantConfigurationListener.name)
+    mongoDatabase.onConnect(SHOW_STYLE_VARIANT_CONFIGURATION_COLLECTION_NAME, () => this.listenForChanges())
+  }
+
+  private listenForChanges(): void {
+    const options: ChangeStreamOptions = { fullDocument: 'updateLookup' }
+    const changeStream: ChangeStream = this.getCollection().watch<ShowStyleVariant, ChangeStreamDocument<ShowStyleVariant>>([], options)
+    changeStream.on('change', () => this.onChange())
+    this.logger.debug('Listening for ShowStyleVariantConfiguration collection changes...')
+  }
+
+  private onChange(): void {
+    // We just want to notify that a changed has been made
```

Comment:
`changed` => `change`.
And why is it that we only care about the event type?

Reply:
Updated.

Because we have just need to notify that a change has happened, so we can update the Configuration. We update the entire configuration, so we don't need special treatment for the exact thing that was updated.